lib/astutils.cpp

```diff
@@ -31,7 +31,7 @@
 #include <stack>
 
 
-void visitAstNodes(const Token *ast, std::function<ChildrenToVisit(const Token *)> visitor)
+void visitAstNodes(const Token *ast, const std::function<ChildrenToVisit(const Token *)>& visitor)
```

Comment:
I just envisioned that it more or less just contained a pointer.. so would be cheap to copy by value.

Reply:
ok... I measured the difference in a small program. A billion function calls:

passing by value took 6.930 seconds

passing by reference too 6.899 seconds

I could see a small performance difference. I used the -O2 flag.

Reply:
Usually, we pass a lambda into this function which needs to be converted to `std::function`. So passing by reference vs passing by value shouldn't make that much of a difference, since a temporary object is always created.

Reply:
For information... this is my testprogram:

```
#include <iostream>
#include <functional>

#include <ctime>

int count;

void dostuff(std::function<int(int)> f)
{
    if (f(std::time(NULL)) > 0)
        ++count;
}

int main() {
    auto f = [=](int x) {return x;};

    for (int i = 0; i < 1000000000; ++i)
        dostuff(f);
    return 0;
}

```